enterprise_gateway/itests/src/itest_notebook.py

```diff
@@ -55,21 +59,28 @@ def execute_and_assert(self, nb_code_entity):
                 test_output = test_code_cell_list[index]
                 self.assertIsNotNone(test_output)
                 assert_code = NotebookTestCase.get_assert_code(real_code_cell.get_first_line_code())
-                test_output = test_output.seralize_output()
-                real_output = real_code_cell.seralize_output()
+                test_output_str = test_output.seralize_output()
+                real_output_str = real_code_cell.seralize_output()
                 try:
                     if assert_code != 2:
                         if assert_code == 0:
-                            self.assertEqual(test_output, real_output)
+                            self.assertEqual(test_output_str, real_output_str)
                         elif assert_code == 1:
-                            self.assertNotEqual(test_output, real_output)
+                            self.assertNotEqual(test_output_str, real_output_str)
+                        elif assert_code == 3 and self.impersonation_username is not None:
+                            # Do impersonation test if and only if the first line is IMPERSONATION (assert code = 3)
+                            # and self.impersonation_username is not None
+                            test_username = test_output.code_output_list[0].raw_output.get('text')
+                            self.assertIsNotNone(test_username)
+                            test_username = str(test_username).replace("\r\n", "")
```

Comment:
I'm surprised this call to `replace()` isn't necessary for the `assert_code == 0` case.  Why can't you just do the comparison against `test_output_str`?  Did you find that `whoami` always concatenates `\r\n`?

We might want to split the replace call into two, one for `\r` and another for `\n`, because I could see this behavior being different depending on the flavor of the OS.